### PR TITLE
PR tweaks for namespace permissions

### PR DESCRIPTION
**Pull request by davidangb**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  - Acceptance criteria exists and is met
  - Note any changes to implementation from the description
  - Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala [hunk 1]
```diff
@@ -63,14 +67,7 @@ object MethodRepository {
...
   def validatePublicOrEmail(email:String): Boolean = {
-    val emailRegex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
-    email match {
-      case null => false
-      case x if x.trim.isEmpty => false
-      case x if emailRegex.findFirstMatchIn(x).isDefined => true
-      case x if x.equalsIgnoreCase("public") => true
-      case _ => false
-    }
+    "public".equals(email) || Try(new InternetAddress(email).validate()).isSuccess
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala:70]: I'm glad we finally have email address validation, but I wasn't super psyched about maintaining our own regex, even if it was copied from a best-practice somewhere. What about this, using the javamail standard?

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/NamespaceApiServiceSpec.scala [hunk 2]
```diff
@@ -31,7 +31,6 @@ class NamespaceApiServiceSpec extends BaseServiceSpec with NamespaceApiService {
             Get(url) ~> dummyUserIdHeaders("1234") ~> sealRoute(namespaceRoutes) ~> check {
               status should equal(OK)
               val permissions = responseAs[List[FireCloudPermission]]
-              permissions shouldNot be (None)
               permissions should be (fcPermissions)
```
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/NamespaceApiServiceSpec.scala:34]: the `shouldNot be (None)` is superfluous - if permissions is None, `permissions should be (fcPermissions)` will fail anyway.

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala (hunk 1 above)
Comment (rushtong) [reply src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala:70]: I'm good with a library version. The version I used was what the W3C recommends.

**Comment by rushtong** [comment]
Ugh, there's a runtime error with this that I didn't catch before the merge

```
app_1    | app[ERROR] java.lang.NoClassDefFoundError: com/sun/mail/util/PropUtil
app_1    | app[ERROR]   at javax.mail.internet.InternetAddress.<clinit>(InternetAddress.java:85)
app_1    | app[ERROR]   at org.broadinstitute.dsde.firecloud.model.MethodRepository$$anonfun$validatePublicOrEmail$1.apply$mcV$sp(MethodRepository.scala:70)
```

I'll dig into it.

**Pull request merged**